### PR TITLE
Check Replicator Activity level

### PR DESCRIPTION
**Pull request by Sandychuang8**

### CODE REVIEW

File: src/Couchbase.Lite.Shared/API/Sync/Replicator.cs [hunk 1]
```diff
@@ -246,12 +246,17 @@ public void Start(bool reset)
...
-                    Config.Database.AddActiveStoppable(this);
                     status = Native.c4repl_getStatus(_repl);
+                    if (status.level == C4ReplicatorActivityLevel.Stopped
+                    || status.level == C4ReplicatorActivityLevel.Stopping
+                    || status.level == C4ReplicatorActivityLevel.Offline) {
+                        ServerCertificate = null;
+                        WriteLog.To.Sync.I(Tag, $"{this}: Starting");
+                        Native.c4repl_start(_repl, Config.Options.Reset || reset);
+                        Config.Options.Reset = false;
+                        Config.Database.AddActiveStoppable(this);
```
Comment (Sandychuang8) [inline src/Couchbase.Lite.Shared/API/Sync/Replicator.cs:257]: Actually AddActiveStoppable is no-op if the replicator is already in ActiveStoppable..

**Review by borrrden — COMMENTED** [review]
What problem is this trying to solve?

**Comment by Sandychuang8** [comment]
The test hanging with infinite "Taking a while for active items to stop..." 
Although I don't see this issue that often anymore lately.

Yesterday I added this test, and I see that hanging error again, but with the latest LiteCore updates, I cannot repro it anymore...

` var config = CreateConfig(true, true, false);
            var replicator = new Replicator(config);

            LoadDocs();
            replicator.Start();
            LoadDocs();
            replicator.Start();
            replicator.Start();
            LoadDocs();
            replicator.Start();
            replicator.Start();
            LoadDocs();
            replicator.Start();`

File: src/Couchbase.Lite.Shared/API/Sync/Replicator.cs [hunk 2]
```diff
@@ -246,12 +246,17 @@ public void Start(bool reset)
...
-                    ServerCertificate = null;
-                    WriteLog.To.Sync.I(Tag, $"{this}: Starting");
-                    Native.c4repl_start(_repl, Config.Options.Reset || reset);
-                    Config.Options.Reset = false;
-                    Config.Database.AddActiveStoppable(this);
                     status = Native.c4repl_getStatus(_repl);
+                    if (status.level == C4ReplicatorActivityLevel.Stopped
+                    || status.level == C4ReplicatorActivityLevel.Stopping
+                    || status.level == C4ReplicatorActivityLevel.Offline) {
+                        ServerCertificate = null;
```
Comment (Sandychuang8) [inline src/Couchbase.Lite.Shared/API/Sync/Replicator.cs:253]: But this change can prevent ServerCertificate set to null during replication and user call replicator Start().

**Review by borrrden — APPROVED** [review]

**Pull request merged**